### PR TITLE
Fix for Cucumber export JSON have missing Property's #60

### PR DESCRIPTION
**Pull request by bitsmakerde**
I added the support for the missing prompters for the JSON export.

I added a JSONSchema check and add a Cucumber JSON Schema to check is the export conform to it

### CODE REVIEW

**Comment by Tyler-Keith-Thompson** [comment]
Thanks for the MR! This looks good, I may have a couple comments.

File: Tests/CucumberSwiftTests/Reporter/Schema.json [hunk 1]
```diff
@@ -0,0 +1,245 @@
+{
```
Comment (Tyler-Keith-Thompson) [inline Tests/CucumberSwiftTests/Reporter/Schema.json:1]: Is this a JSONSchema representation of the proper Cucumber JSON report? Love this, thank you!

File: Sources/CucumberSwift/Gherkin/Parser/Feature.swift [hunk 2]
```diff
@@ -67,7 +67,7 @@ public class Feature: Taggable, Positionable {
         return false
     }
 
-    func toJSON() -> [String: Any] {
+    /*func toJSON() -> [String: Any] {
```
[the remark this answers is not in the document]
Comment (bitsmakerde) [reply Sources/CucumberSwift/Gherkin/Parser/Feature.swift:70]: This function was not used by you. I commit it out to test, is it is use. I would say let's remove not used code

File: Sources/CucumberSwift/Reporter/JSONReporter.swift [hunk 3]
```diff
@@ -86,13 +86,17 @@ extension CucumberJSONReporter {
         let description: String
         let keyword: String = "Feature"
         var elements: [Scenario] = []
+        let line: UInt
```
[the remark this answers is not in the document]
Comment (bitsmakerde) [reply Sources/CucumberSwift/Reporter/JSONReporter.swift:89]: mh ok. But Lexer not comforts to encodable. I ask me, is this a work we should to to conform or create a sub Struct which conforms and to map to this.

`Type 'CucumberJSONReporter.Feature' does not conform to protocol 'Encodable'`

File: Tests/CucumberSwiftTests/Reporter/ReporterTests.swift [hunk 4]
```diff
@@ -132,6 +133,38 @@ class ReporterTests: XCTestCase {
...
+                    Given(I: print(""))
+                }
+            }
+            reporter.testSuiteStarted(at: Date())
+            Cucumber.shared.executeFeatures()
+            let actual = try XCTUnwrap(try JSONSerialization.jsonObject(with: JSONEncoder().encode(reporter.features)) as? [[AnyHashable: Any]])
+            let result = try JSONSchema.validate(actual, schema: schema)
+            if let errors = result.errors {
+                errors.forEach { error in
+                    print(error.instanceLocation)
```
[the remark this answers is not in the document]
Comment (bitsmakerde) [reply Tests/CucumberSwiftTests/Reporter/ReporterTests.swift:158]: Yes you are right. I changed it.

**Comment by bitsmakerde** [comment]
I commit the changes.

**Pull request merged**